src/main/java/io/github/treesitter/jtreesitter/Query.java

```diff
@@ -503,11 +507,24 @@ public Stream<QueryMatch> findMatches(Node node) {
      * @param node The node that the query will run on.
      * @param predicate A function that handles custom predicates.
      */
-    public Stream<QueryMatch> findMatches(Node node, @Nullable BiPredicate<QueryPredicate, QueryMatch> predicate) {
+    public Stream<QueryMatch> findMatches(Node node, @Nullable BiPredicate<QueryPredicate, QueryMatch> predicate){
+        return findMatches(node, predicate, arena);
+    }
+
+
+    /**
+     * Like {@link #findMatches(Node, BiPredicate)} but the native memory of the returned matches is created using the
+     * given allocator.
+     *
+     * @param node The node that the query will run on.
+     * @param predicate A function that handles custom predicates.
+     * @param allocator The allocator that is used to allocate the native memory of the returned matches.
+     */
+    public Stream<QueryMatch> findMatches(Node node, @Nullable BiPredicate<QueryPredicate, QueryMatch> predicate, SegmentAllocator allocator) {
```

Comment:
[Kotlin convention](https://kotlinlang.org/docs/lambdas.html#passing-trailing-lambdas)
```suggestion
    public Stream<QueryMatch> findMatches(Node node, SegmentAllocator allocator, @Nullable BiPredicate<QueryPredicate, QueryMatch> predicate) {
```

Reply:
Changed